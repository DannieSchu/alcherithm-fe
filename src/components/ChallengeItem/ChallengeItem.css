.ChallengeItem {
  padding: 2rem;

  div {
    display: flex;
    justify-content: space-between;
    width: 95%;
  }
}
