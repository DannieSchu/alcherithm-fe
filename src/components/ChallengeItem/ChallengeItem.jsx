import React from 'react';
import PropTypes from 'prop-types';
import styles from './ChallengeItem.css';

const ChallengeItem = ({ category, challengeNumber, instructions, solutions, char }) => {
  const slicedInstructions = `${instructions.slice(0, char)}...`;
  const icon = solutions[0].passed ? '✔' : '✖️';

  return (
    <section className={styles.ChallengeItem}>
      <h3>{category} {challengeNumber}</h3>
      <p>{slicedInstructions}</p>
      {solutions[0].date && <div><h4>{icon}</h4><h4>{solutions[0].date}</h4></div>}
    </section>
  );
};

ChallengeItem.propTypes = {
  category: PropTypes.string.isRequired,
  challengeNumber: PropTypes.number.isRequired,
  instructions: PropTypes.string.isRequired,
  solutions: PropTypes.arrayOf(PropTypes.shape({
    passed: PropTypes.boolean,
    date: PropTypes.string
  })),
  char: PropTypes.number.isRequired
};

export default ChallengeItem;
