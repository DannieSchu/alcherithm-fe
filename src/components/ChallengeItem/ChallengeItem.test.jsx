import React from 'react';
import { shallow } from 'enzyme';
import ChallengeItem from './ChallengeItem';

describe('ChallengeItem component', () => {
  it('renders ChallengeItem', () => {
    const wrapper = shallow(<ChallengeItem 
      category='forEach' 
      challengeNumber={1} 
      instructions='Write a function named greeting that takes in a string and returns the string in all uppercase letters. Then, write a function named speaker that takes in a string and a callback function. The speaker function should return the string in all uppercase letters only by invoking the callback.' 
      solutions={[{ passed: true, date: '03/02/20' }]}
      char={100}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
