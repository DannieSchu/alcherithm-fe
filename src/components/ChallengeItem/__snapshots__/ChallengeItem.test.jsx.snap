// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeItem component renders ChallengeItem 1`] = `
<section
  className="ChallengeItem"
>
  <h3>
    forEach
     
    1
  </h3>
  <p>
    Write a function named greeting that takes in a string and returns the string in all uppercase lette...
  </p>
  <div>
    <h4>
      âœ”
    </h4>
    <h4>
      03/02/20
    </h4>
  </div>
</section>
`;
